sdk/core/azure-core-amqp/src/impl/rust_amqp/amqp/session.cpp

```diff
@@ -102,23 +102,24 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
 
     if (m_options.MaximumLinkCount.HasValue())
     {
-      amqpsessionoptionsbuilder_set_handle_max(
-          optionsBuilder.get(), m_options.MaximumLinkCount.Value());
+      optionsBuilder.reset(amqpsessionoptionsbuilder_set_handle_max(
```

Comment:
if `x.reset(f(x.release(), ...))` becomes a very, very common pattern when interoping with Rust, I think we can write some C++ syntactic sugar to help with that.

Reply:
Oops - I missed this one when I was making my conversion to use `invoke_builder_api`

Reply:
And I totally missed the PR description, that literally a function that I am proposing to write one day was one of the goals of this very PR :)